### ISSUE TITLE
Layout/HeredocIndentation only works for toplevel code

### ISSUE THREAD

**Issue by Epigene** [rubocop/rubocop#10478]
As-is `Layout/HeredocIndentation` just does not work, it fails at the two things it could be useful for - enforcing squiggly heredoc use and proper indentation of doc bodies relative to delimiters, plus, the offence message makes no sense.

From the description of `Layout/HeredocIndentation`:

> This cop checks the indentation of the here document bodies. The bodies are indented one step.

A sample offence message

> Layout/HeredocIndentation: Use 2 spaces for indentation in a heredoc by using <<~ instead of <<-.

--------

## Steps to reproduce the problem

Configure rubocop
```
AllCops:
  RubyInterpreters:
    - ruby
  Include:
    - '**/*.rb'
  TargetRubyVersion: 2.7
  DisabledByDefault: true

Layout/HeredocIndentation:
  Enabled: true

Layout/ClosingHeredocIndentation:
  Enabled: true
```

Set up the example file
```rb
<<-AA
  xx
AA

<<-BB
   xx
BB

module MyModule
  <<-CC
xx
  CC

  <<-DD
  xx
  DD

  <<-EE
    xx
  EE
  
  def m1
    <<-FF
    xx
    FF
  end  

  def m2
    <<-GG
      xx
    GG
  end 
end
```

## Expected behavior

Naively, one would expect these offences:
1. line 6 has excess indent
2. line 11 has missing indent (works, kinda)
3. line 15 is missing 2 spaces of indent
4. line 24 is missing 2 spaces of indent

## Actual behavior

Only one offence is reported

```
test.rb:11:1: C: [Correctable] Layout/HeredocIndentation: Use 2 spaces for indentation in a heredoc by using <<~ instead of <<-.
xx # expect failure for bad indent, should be 2 spaces relative to opening delimiter (4 in total) ...
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

## RuboCop version

```
$ bundle exec rubocop -V
1.26.1 (using Parser 3.1.1.0, rubocop-ast 1.16.0, running on ruby 2.7.5 x86_64-darwin20)
```

## Suggested changes

1. Would be great to get offences for each non-squiggly heredoc use, maybe the clean approach is to have a separate cop for that
2. Make doc body indent be checked properly relative to opening delimiter
3. Splitting squiggly-ness cop from indent cop would also clear up the confusing offence message problem

```
class MyClass
  def method_name
    <<-BB # expect failure for missing squiggly
   xx # expect failure for bad indent, should be at at least 2 spaces deeper than opening delimiter (6+)
    BB
  end
end
```

Would produce two offences:
```
test.rb:3:3: C: [Correctable] Layout/SquigglyHeredoc: Use squiggly heredoc <<~ instead of regular <<-.
<<-BB
  ^

test.rb:4:1: C: [Correctable] Layout/HeredocIndentation: Use at least 2 spaces of indentation for heredoc body relative to heredoc opening delimiter 
   xx # expect failure for bad indent, should be at at least 2 spaces deeper than opening delimiter (6+)
^^^^^^
```

4. (addendum) May even go the extra mile and cop that no subsequent line after the first line of a heredoc body has indent that is less than the first line's indent. For example

```rb
class MyClass
  def method_name
    <<-CC # expect failure for missing squiggly
       line1 # uses one excess space, but we can't do anything about that
      line2 # offensive, given first line's indent
    CC
  end
end
```

**Comment by dvandersluis** [comment]
Without a squiggly heredoc (ie. for `<<` or `<<-`), ruby starts the string from column 0, therefore:

```ruby
  <<-CC
xx
  CC
=> "xx\n"

  <<-DD
  xx
  DD
=> "  xx\n"
```

Moving the indentation of one of these heredocs would change the value of the string. This is why squiggly heredocs were added I believe, so that the heredoc itself can be indented without affecting the contents of the string.

All of your example heredocs therefore have a different amount of leading spaces. Correcting the `CC` heredoc is safe because the leading spaces don't change (because the original heredoc content starts in column 0).

None of the other heredocs could be replaced with squiggly heredocs without changing the resulting value.
